### PR TITLE
[FEATURE] Support javaScriptModules in FormEngine resultArray

### PR DESCRIPTION
**Pull request by linawolf**
resolves https://github.com/TYPO3-Documentation/Changelog-To-Doc/issues/209

### CODE REVIEW

File: Documentation/ApiOverview/Backend/ContextualMenu.rst [hunk 1]
```diff
@@ -179,23 +179,15 @@ Enabling context menu in your own backend modules is quite straightforward.
 The examples below are taken from the "beuser" system extension and
 assume that the module is Extbase-based.
 
-..  todo: Document the new ES6 way of creating a context menu
-    https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-CoreApi/issues/2298
-
 The first step is to include the needed JavaScript using the
-:js:`includeRequireJsModules` property
+:js:`includeJavaScriptModules` property
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Backend/ContextualMenu.rst:183]:
```suggestion
:html:`includeJavaScriptModules` property
```

File: Documentation/ApiOverview/FormEngine/Rendering/_ImportDataControl.php [hunk 2]
```diff
@@ -18,6 +18,8 @@ public function render(): array
                 'class' => 'importData ',
                 'data-id' => $this->data['databaseRow']['somefield'],
             ],
+            'javaScriptModules' => ['@my_vendor/my_extension/import-data.js'],
```
Comment (brotkrueml) [inline Documentation/ApiOverview/FormEngine/Rendering/_ImportDataControl.php:21]: As far as I understand: now in v12 both JS files are included: The ES6 and RequireJS one.

So, when using the javaScriptModules in combination with requireJsModules, we need a version check.

**Review by brotkrueml — CHANGES_REQUESTED** [review]

File: Documentation/ApiOverview/FormEngine/Rendering/_ImportDataControl.php (hunk 2 above)
Comment (linawolf) [reply Documentation/ApiOverview/FormEngine/Rendering/_ImportDataControl.php:21]: As far as I understand this we are just providing information here so we can provide both?

Comment (brotkrueml) [reply Documentation/ApiOverview/FormEngine/Rendering/_ImportDataControl.php:21]: Okay,  looked up the code, resulting in this example to add both: the ES6 and RequireJS JavaScripts:
https://github.com/TYPO3/typo3/blob/main/typo3/sysext/backend/Classes/Form/FormResultCompiler.php#L207-L213

Comment (linawolf) [reply Documentation/ApiOverview/FormEngine/Rendering/_ImportDataControl.php:27]: done

**Review by brotkrueml — APPROVED** [review]

**Pull request merged**